src/main/java/com/feywild/feywild/world/gen/OreType.java

```diff
@@ -7,64 +7,85 @@
 import net.minecraft.core.Registry;
 import net.minecraft.data.BuiltinRegistries;
 import net.minecraft.data.worldgen.features.OreFeatures;
-import net.minecraft.util.valueproviders.ConstantInt;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.levelgen.VerticalAnchor;
 import net.minecraft.world.level.levelgen.feature.Feature;
 import net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration;
 import net.minecraft.world.level.levelgen.heightproviders.HeightProvider;
 import net.minecraft.world.level.levelgen.heightproviders.UniformHeight;
-import net.minecraft.world.level.levelgen.placement.CountOnEveryLayerPlacement;
 import net.minecraft.world.level.levelgen.placement.HeightRangePlacement;
 import net.minecraft.world.level.levelgen.placement.PlacedFeature;
 
 import java.util.Objects;
 import java.util.function.Supplier;
 
 public enum OreType {
-    FEY_GEM_ORE(ModBlocks.feyGemBlock,
+    FEY_GEM_ORE(ModBlocks.feyGemBlock, ModBlocks.feyGemBlockDeepSlate,
             ModBlocks.feyGemBlockLivingrock,
             () -> WorldGenConfig.ores.fey_gem
     );
 
-    private final Block block;
+    private final Block stoneBlock;
+    private final Block deepSlateBlock;
     private final Block alfheimBlock;
     private final Supplier<OreData> data;
-    private final LazyValue<PlacedFeature> feature;
+
+    private final LazyValue<PlacedFeature> stoneFeature;
+    private final LazyValue<PlacedFeature> deepSlateFeature;
     private final LazyValue<PlacedFeature> alfheimFeature;
 
     @SuppressWarnings("deprecation")
-    OreType(Block block, Block alfheimBlock, Supplier<OreData> data) {
-        this.block = block;
+    OreType(Block feyGemBlock, Block feyGemBlockDeepSlate, Block alfheimBlock, Supplier<OreData> data) {
+        this.stoneBlock = feyGemBlock;
+        this.deepSlateBlock = feyGemBlockDeepSlate;
         this.alfheimBlock = alfheimBlock;
         this.data = data;
-        this.feature = new LazyValue<>(() -> {
-            OreConfiguration oreFeatureConfig = new OreConfiguration(OreFeatures.NATURAL_STONE, block.defaultBlockState(), getMaxVeinSize());
-            return Registry.register(BuiltinRegistries.PLACED_FEATURE, Objects.requireNonNull(block.getRegistryName()), Feature.ORE.configured(oreFeatureConfig).placed(
-                    HeightRangePlacement.triangle(VerticalAnchor.absolute(getMinHeight()), VerticalAnchor.absolute(getMaxHeight())),
-                    CountOnEveryLayerPlacement.of(ConstantInt.of(getSpawnWeight()))
-            ));
+
+        this.stoneFeature = new LazyValue<>(() -> {
```

Comment:
Create a utility method to create these `PlacedFeatures`. Also it might be good to make it that a method reference to `OrePlacement::commonOrePlacement` is passed to the constructor and then used in the placements, so other ores could use rare placements if needed.

Reply:
I hope I did this correctly. Otherwise you might have to assist.

Reply:
The `OrePlacement::commonOrePlacement` is still not passed through the constructor, but I can fix that.